openmdao/components/tests/test_subproblem_comp.py

```diff
@@ -0,0 +1,37 @@
+from numpy import pi
+
+import openmdao.api as om
+from openmdao.utils.assert_utils import assert_near_equal, assert_check_partials, assert_check_totals
+import unittest
+
+from openmdao.components.subproblem_comp import SubproblemComp
+
+class TestSubproblemComp(unittest.TestCase):
+    def test_subproblem_comp(self):
+        prob = om.Problem()
+
+        model = om.ExecComp('z = x**2 + y')
+        submodel1 = om.ExecComp('x = r*cos(theta)')
```

Comment:
There have been discussions recently about deprecating the ability to specify a component as the problem model, so you may want to avoid having to go back and update this test in the future by just making the subproblem models be Groups instead of ExecComps.  Using Groups, and potentially nested ones, in the subproblem models is probably a good idea anyway, in order to test that the name mapping/promotion stuff is working correctly.